src/adsbinarynode.jl

```diff
@@ -0,0 +1,400 @@
+# This file is part of the TaylorIntegration.jl package; MIT licensed
+
+using StaticArrays: SVector
+import AbstractTrees: HasNodeType, NodeType, ParentLinks, StoredParents, TreeIterator,
+                      children, nodetype, nodevalue, parent, printnode, PreOrderDFS
+import Base: IteratorEltype, HasEltype, in, eltype, split
+
+if !isdefined(Base, :isnothing)        # Julia 1.0 support
+    using AbstractTrees: isnothing
+end
+
+export ADSDomain
+
+"""
+    ADSDomain{N, T <: Real}
+
+`N` dimensional box used in Automatic Domain Splitting to represent the
+domain of `N` jet transport variables.
+
+# Fields
+
+- `lo::SVector{N, T}`: lower bounds.
+- `hi::SVector{N, T}`: upper bounds.
+"""
+struct ADSDomain{N, T <: Real}
+    lo::SVector{N, T}
+    hi::SVector{N, T}
+    # Inner constructor
+    function ADSDomain{N, T}(lo::SVector{N, T}, hi::SVector{N, T}) where {N, T <: Real}
+        @assert all(hi .> lo) "Each upper bound must be greater than the respective lower bound"
+        new{N, T}(lo, hi)
+    end
+end
+
+# Outer constructors
+ADSDomain(lo::SVector{N, T}, hi::SVector{N, T}) where {N, T <: Real} = ADSDomain{N, T}(lo, hi)
+
+function ADSDomain(x::Tuple{T, T}...) where {T <: Real}
+    N = length(x)
+    return ADSDomain{N, T}(
+        SVector{N, T}(x[i][1] for i in eachindex(x)),
+        SVector{N, T}(x[i][2] for i in eachindex(x))
+    )
+end
```

Comment:
Testing locally, I obtain that this method is not passing an Aqua test, more specifically, 
```julia
ua = Aqua.detect_unbound_args_recursively(TaylorIntegration)
```
signals this method. (`ua` should have length 0.)

Reply:
Following Aqua's documentation, I added a trivial constructor `ADSDomain()`.

Reply:
Regarding this point (unless constructing an `ADSDomain` without arguments has a specific use), I would be in favor of `ADSDomain()` returning an error? Simply because throughout the code the constructor with arguments is the one that will be used to construct `ADSDomain` variables, and if by any chance we call the wrong constructor we might end up with silent errors / hard-to-catch bugs.